core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -226,6 +228,74 @@ public void cancelOperation() throws UnsupportedOperationException {
         return this.fluxHandle.blockLast();
     }
 
+    /**
+     * Blocks until given {@link OperationStatus} is received or a timeout expires.
+     * @param statusToBlockFor The {@link OperationStatus} to block for and it can be any valid {@link OperationStatus} except {@link OperationStatus#OTHER}.
+     * @param timeout The time after which it will stop blocking.
+     * @return
+     */
+    public PollResponse<T> blockUtil(PollResponse.OperationStatus statusToBlockFor, Duration timeout) {
+        if (statusToBlockFor == null || statusToBlockFor == PollResponse.OperationStatus.OTHER) {
+            throw new IllegalArgumentException("The status input parameter can not be null or OTHER.");
+        }
+        Objects.requireNonNull(timeout, "The timeout input parameter cannot be null.");
+        if (timeout.toNanos() <= 0) {
+            throw new IllegalArgumentException("Negative or zero value for timeout not allowed.");
+        }
+
+        if (!isAutoPollingEnabled()) {
+            setAutoPollingEnabled(true);
+        }
+        return this.fluxHandle.filter(tPollResponse -> matchStatus(tPollResponse, null, statusToBlockFor)).blockFirst(timeout);
```

Comment:
Should the timeout be conditional? In other SDKs it has been optional where if it is null there is no timeout

Reply:
We should be consistent , I think, across  all places. @JonathanGiles  . I have overloaded method 
blockUntil(PollResponse.OperationStatus statusToBlockFor) 
blockUntil(PollResponse.OperationStatus statusToBlockFor, Duration timeout)